src/services/data/fetchCategories.js

```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+
+export const fetchCategories = async ({ videoStateDispatch }) => {
+    try {
+        const res = await axios.get("/api/categories");
+        videoStateDispatch({ type: "ADD_CATEGORIES", payload: res.data.categories });
```

Comment:
you should check the status first and then run the dispatch method, also follow this in the other api calls

Reply:
thanks, done